apps/app_voicemail.c

```diff
@@ -12771,20 +13135,26 @@ static char *complete_voicemail_show_users(const char *line, const char *word, i
 	int wordlen;
 	struct ast_vm_user *vmu;
 	const char *context = "";
+	char *ret = NULL;
 
 	/* 0 - voicemail; 1 - show; 2 - users; 3 - for; 4 - <context> */
 	if (pos > 4)
 		return NULL;
 	wordlen = strlen(word);
+	AST_LIST_LOCK(&users);
 	AST_LIST_TRAVERSE(&users, vmu, list) {
 		if (!strncasecmp(word, vmu->context, wordlen)) {
-			if (context && strcmp(context, vmu->context) && ++which > state)
-				return ast_strdup(vmu->context);
+			if (context && strcmp(context, vmu->context) && ++which > state) {
+				ret = ast_strdup(vmu->context);
+				AST_LIST_UNLOCK(&users);
+				return ret;
+			}
 			/* ignore repeated contexts ? */
 			context = vmu->context;
 		}
 	}
-	return NULL;
+	AST_LIST_UNLOCK(&users);
+	return ret;
```

Comment:
I'm not following why you can't just return NULL here. In which case the initialize of ret to NULL is also unnecessary.